src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -411,6 +405,24 @@ public void deleteDataset(Integer datasetId, Integer userId) throws Exception {
         ).collect(Collectors.toList());
     }
 
+    private boolean filterDatasetOnProperties(DatasetDTO dataset, String term) {
+        //datasets need to have consentId, null check to prevent NPE
+        Boolean consentIdMatch;
+        String consentId = dataset.getConsentId();
+        if(Objects.isNull(consentId)) {
+            consentIdMatch = false;
+        } else {
+            consentIdMatch = consentId.toLowerCase().contains(term);
+        }
+        return dataset.getProperties()
+        .stream()
+        .filter(p -> Objects.nonNull(p.getPropertyValue()))
+        .anyMatch(p -> {
+            return consentIdMatch
+            || p.getPropertyValue().toLowerCase().contains(term);
+        });
```

Comment:
Minor nit: indentation. Not sure what your IDE settings are, but it should indent 4 or 8 spaces. If your IDE let's you use [Google's Java Style](https://medium.com/swlh/configuring-google-style-guide-for-java-for-intellij-c727af4ef248), see if you can select that and it should format your code automatically.